src/annotator/guest.js

```diff
@@ -499,21 +509,26 @@ export default class Guest {
     this._updateAnchors(this.anchors.concat(anchors), true /* notify */);
 
     // Let other frames (eg. the sidebar) know about the new annotation.
-    this._annotationSync.sync([annotation]);
+    this._sidebarRPC.call('syncAnchoringStatus', annotation);
 
     return anchors;
   }
 
   /**
    * Remove the anchors and associated highlights for an annotation from the document.
    *
-   * @param {AnnotationData} annotation
+   * @param {string} tag
    * @param {boolean} [notify] - For internal use. Whether to emit an `anchorsChanged` notification
    */
-  detach(annotation, notify = true) {
+  detach(tag, notify = true) {
+    if (!this._anchoringCompleted) {
```

Comment:
As noted above, a boolean variable won't be sufficient here.

Reply:
Yes, it is clear, thanks.